src/components/CookieProvider.ts

```diff
@@ -0,0 +1,27 @@
+import { Ajax } from 'src/libs/ajax';
+import { AbortOption } from 'src/libs/ajax/data-provider-common';
+import { azureCookieReadyStore, cookieReadyStore } from 'src/libs/state';
+
+export interface CookieProvider {
+  invalidateCookies: (options?: AbortOption) => Promise<void>;
+}
+export const cookieProvider: CookieProvider = {
+  invalidateCookies: async (options: AbortOption = {}) => {
+    const { signal } = options;
+    const cookies: string[] = document.cookie.split(';');
+    // TODO: call azure invalidate cookie once endpoint exists, https://broadworkbench.atlassian.net/browse/IA-3498
+    await Ajax(signal)
+      .Runtimes.invalidateCookie()
+      .catch(() => {});
+    // Expire all cookies
```

Comment:
Do we want to expire all cookies? Feels like we just want to expire `LeoToken`, no?

And not sure this is necessary for `LeoToken` -- the `Runtimes.invalidateCookie()` request should return a `Set-Cookie` response header which invalidates it.

Reply:
I did think this was odd, but there aren't any tests or comments to indicate why this is done, so I'm a bit inclined to not mess with it and keep it as is since its used elsewhere when interacting with cookies